File: neetcode all and misc questions/Arrays and Strings/majority_element_ii.py
class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        """
        O(N) Time O(N) Space with counter,
        just pick the elements that have a frequency > N//3 

        The O(1) Space solution involves an algorithm called
        Boyer-Moore Voting Algorithm. It doesn't seem like something
        that would be asked in an interview but the code for that is here:

        https://youtu.be/Eua-UrQ_ANo
        """
        c = Counter(nums)
        factor = len(nums)//3
        return [key for key in c if c[key] > factor]

File: node_js_ds_and_algos_implementations/arrays and strings/hashmaps/majority_element_ii.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    majorityElement(nums) {
        let counter = new Map()
        for (let i = 0; i < nums.length; ++i){
            counter.set(nums[i], (counter.get(nums[i]) || 0) + 1)
        } 
        let res = []
        for (const [key, value] of counter.entries()){
            if (value > Math.floor(nums.length/3)){
                res.push(key)
            }
        }
        return res
    }
}
